web/src/unread_ui.ts

```diff
@@ -23,8 +23,12 @@ export function register_update_unread_counts_hook(f: UpdateUnreadCountsHook): v
     update_unread_counts_hooks.push(f);
 }
 
-function set_mark_read_on_scroll_state_banner(template: string): void {
-    $("#mark_read_on_scroll_state_banner").html(template);
+function set_mark_read_on_scroll_state_banner(rendered_banner: string): void {
+    $("#mark_read_on_scroll_state_banner").html(rendered_banner);
+    // This place holder is essential since hiding the unread banner would reduce
+    // scrollable place, shifting the message list downward if the scrollbar is
+    // at bottom.
+    $("#mark_read_on_scroll_state_banner_place_holder").html(rendered_banner);
```

Comment:
Why does the same HTML go into both places? I don't really understand.

Reply:
>To avoid this we can add a mark_read_on_scroll_state_banner_place_holder div below typing notification div which just has the same dimesnsions as the unread banner before it is hidden. 

According to Aman's comment, we want same dimensions both the actual banner and the placeholder banner. So both of them are being rendered with the same template so that the dimensions are adjusted automatically.